javatests/arcs/core/allocator/AllocatorTestBase.kt

```diff
@@ -132,6 +135,12 @@ open class AllocatorTestBase {
         WritePerson.throws = false
     }
 
+    @After
+    fun tearDown() {
+        schedulerProvider.cancelAll()
+        Thread.sleep(1000)
```

Comment:
When and why do we need a sleep like this?

If it's for waiting all jobs done on its scheduler, we should wait and complete all jobs while responding to the cancelAll() w/ or w/out a timeout.

Reply:
I agree completely. I've been fighting with flakiness in this test class while trying to get this PR to pass. I'll just start @Ignore'ing the flakes.  I've got a task when I get time to work on it to audit all of our coroutine launches and determine how to make everything cancelable and scoped correctly.

Reply:
Perfect!